sdk/core/Azure.Core/src/RequestFailedException.cs

```diff
@@ -258,6 +259,7 @@ internal static bool TryExtractErrorContent(Response response, out ResponseError
                     return false;
                 }
                 error = System.Text.Json.JsonSerializer.Deserialize<ErrorResponse>(content)?.Error;
+                error ??= System.Text.Json.JsonSerializer.Deserialize<ResponseError>(content);
```

Comment:
Does Deserialize only throw if it is invalid JSON?

Reply:
Would error be null if the content doesn't match ErrorResponse or would the properties just be empty?

Reply:
> Would error be null if the content doesn't match ErrorResponse or would the properties just be empty?

Nevermind - we are checking if the Error property is null on the deserialized instance, which makes sense.

Reply:
We should consider this along with #36226. Could we, for example, effectively specialize `BinaryData.ToObjectFromJson<ResponseError>` to do this fallback itself?

Reply:
> > Does Deserialize only throw if it is invalid JSON?
> 
> Is there another scenario you are concerned about?

Nope, just wanted to make sure we always fallback when appropriate.